app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/coverage/PerilBottomSheet.kt

```diff
@@ -2,68 +2,204 @@ package com.hedvig.app.feature.insurance.ui.detail.coverage
+                                        }
+                                        defaultSystemUiVisibility?.let {
+                                            dialog?.window?.decorView?.systemUiVisibility = it
+                                        }
+                                        dialog?.window?.addFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND)
+                                    }
+                                    BottomSheetBehavior.STATE_COLLAPSED -> {
+                                        close.setOnClickListener(null)
+                                    }
+                                }
+                            }
 
-            if (titleText == null || bodyText == null || iconUrl == null) {
-                e { "Programmer error: Missing either TITLE, BODY or ICON_URL in ${this.javaClass.name}" }
+                            override fun onSlide(bottomSheet: View, slideOffset: Float) {
+                                close.alpha = slideOffset
+                                shadow.alpha = 1 - slideOffset
+                            }
+                        })
+                }
+            }
+
+            if (title == null || description == null || iconUrl == null || exception == null || covered == null || info == null) {
+                e { "Programmer error: Missing arguments in ${this@PerilBottomSheet.javaClass.name}" }
                 return
             }
 
-            val requestBuilder = buildRequestBuilder()
+            recycler.adapter = CoveredAndExceptionAdapter().also {
+                safeLet(
+                    title,
+                    description,
+                    info,
+                    covered,
+                    exception,
+                    iconUrl
+                ) { t, d, i, c, e, u ->
+                    it.submitList(
+                        expandedList(t, d, i, c, e, u)
+                    )
+                }
+            }
+        }
+    }
+
+    private fun expandSheet() {
+        val parentLayout =
+            dialog?.findViewById<View>(com.google.android.material.R.id.design_bottom_sheet)
 
-            requestBuilder
-                .load(iconUrl)
-                .into(icon)
-            title.text = titleText
-            body.text = bodyText
+        parentLayout?.let { parent ->
+            val behaviour = BottomSheetBehavior.from(parent)
+            setupFullHeight(parent)
+            behaviour.state = BottomSheetBehavior.STATE_EXPANDED
         }
     }
 
+    private fun setupFullHeight(bottomSheet: View) {
+        val layoutParams = bottomSheet.layoutParams
+        layoutParams.height = WindowManager.LayoutParams.MATCH_PARENT
+        bottomSheet.layoutParams = layoutParams
+    }
+
+    private fun expandedList(
+        title: String,
+        description: String,
+        info: String,
+        covered: ArrayList<String>,
+        exceptions: ArrayList<String>,
+        iconLink: String
+    ) = listOfNotNull(
```

Comment:
Which of these can be null? If none, use `listOf()` instead.

Reply:
Dont know why i used this 🤷